services/repository/branch.go

```diff
@@ -642,3 +643,61 @@ func SetRepoDefaultBranch(ctx context.Context, repo *repo_model.Repository, gitR
 
 	return nil
 }
+
+type BranchDivergingInfo struct {
+	BaseIsNewer   bool
```

Comment:
I think the old name `BaseHasNewCommits` is better. Because the "base" might not be really "newer"